x/gamm/simulation/sim_msgs.go

```diff
@@ -297,6 +300,10 @@ func getRandPool(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context) (uint64
 	poolCoins = simtypes.RemoveIndex(poolCoins, index)
 	coinOut := poolCoins[0]
 	poolDenoms := osmoutils.CoinsDenoms(pool.GetTotalPoolLiquidity(ctx))
-	gammDenom := types.GetPoolShareDenom(pool_id)
-	return pool_id, pool, coinIn, coinOut, poolDenoms, gammDenom, err
+
+	// pool ids start from 1.
+	poolId := poolIndex + 1
```

Comment:
if we are max bound by the upcoming pool ID, its possible the poolIndex is the current largest poolID, so if we add one to here wouldn't it be a non existent pool?

Reply:
I think this is right

Assume we have 3 pools with ids: 1, 2, 3.

So, the max `poolIndex` we can generate is 2.

Then, max `poolId` is 3 which is the correct one

Reply:
yeah I think you are correct, the verbiage in the comment "max bound by next poolID" was confusing which is why I had the initial impression